src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -261,12 +267,81 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
             tr.style.height = getComputedStyle(tr2).height;
         });
 
-        // Adjust fixed columns data to data height
-        const contentTd = r1c1.querySelector('tr > td:first-of-type');
-        if (contentTd) {
-            const contentTr = contentTd.parentElement as HTMLElement;
+        if (fixed_columns) {
+            const r1c0Table = r1c0.querySelector('table') as HTMLElement;
+            const r1c1Table = r1c0.querySelector('table') as HTMLElement;
+
+            r1c0Table.style.width = getComputedStyle(r1c1Table).width;
+
+            const lastVisibleTd = r1c0.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns})`);
+
+            let it = 0;
+            let currentWidth = r1c0.getBoundingClientRect().width;
+            let lastWidth = currentWidth;
+
+            do {
+                lastWidth = currentWidth
+
+                // Force first column containers width to match visible portion of table
+                if (lastVisibleTd) {
+                    const r1c0FragmentBounds = r1c0.getBoundingClientRect();
+                    const lastTdBounds = lastVisibleTd.getBoundingClientRect();
+                    currentWidth = lastTdBounds.right - r1c0FragmentBounds.left;
+
+                    const width = `${currentWidth}px`;
+
+                    r0c0.style.width = width;
+                    r1c0.style.width = width;
+                }
+
+                // Force second column containers width to match visible portion of table
+                const firstVisibleTd = r1c1.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns + 1})`);
+                if (firstVisibleTd) {
+                    const r1c1FragmentBounds = r1c1.getBoundingClientRect();
+                    const firstTdBounds = firstVisibleTd.getBoundingClientRect();
+
+                    const width = firstTdBounds.left - r1c1FragmentBounds.left;
+                    r0c1.style.marginLeft = `-${width}px`;
+                    r0c1.style.marginRight = `${width}px`;
+                    r1c1.style.marginLeft = `-${width}px`;
+                    r1c1.style.marginRight = `${width}px`;
+                }
 
-            this.stylesheet.setRule('.dash-fixed-column tr', `height: ${getComputedStyle(contentTr).height};`);
+                it++;
+
+                if (
+                    (Math.abs(currentWidth - lastWidth) < WIDTH_EPSILON && it % 2 === 0) ||
+                    it >= MAX_WIDTH_ITERATIONS
+                ) {
+                    break;
+                }
+            } while (true);
```

Comment:
Not a big deal, but given that ultimately there's a known limit (`MAX_WIDTH_ITERATIONS`) I'd have probably written this as a `for` loop, with only the width test down here breaking the loop early.

Reply:
Want to keep both exit conditions together, that's why I opted for a `while` instead of a `for` loop. Did some clean up so the exit condition is now in the `while` condition instead of the body. https://github.com/plotly/dash-table/pull/722/commits/207c90ab62e2c558a2e9e84d4284532f58d27db1

Reply:
Also removed the `it % 2 == 0` check which is no longer necessary

Reply:
OK, makes sense. Glad the `it % 2 === 0` can be avoided!